tracing-opentelemetry/src/subscriber.rs

```diff
@@ -489,37 +485,37 @@ where
     fn on_record(&self, id: &Id, values: &Record<'_>, ctx: Context<'_, C>) {
         let span = ctx.span(id).expect("Span not found, this is a bug");
         let mut extensions = span.extensions_mut();
-        if let Some(builder) = extensions.get_mut::<otel::SpanBuilder>() {
-            values.record(&mut SpanAttributeVisitor(builder));
+        if let Some(data) = extensions.get_mut::<OtelData>() {
+            values.record(&mut SpanAttributeVisitor(&mut data.builder));
         }
     }
 
     fn on_follows_from(&self, id: &Id, follows: &Id, ctx: Context<C>) {
         let span = ctx.span(id).expect("Span not found, this is a bug");
         let mut extensions = span.extensions_mut();
-        let builder = extensions
-            .get_mut::<otel::SpanBuilder>()
-            .expect("Missing SpanBuilder span extensions");
+        let data = extensions
+            .get_mut::<OtelData>()
+            .expect("Missing otel data span extensions");
 
         let follows_span = ctx
             .span(follows)
             .expect("Span to follow not found, this is a bug");
         let mut follows_extensions = follows_span.extensions_mut();
-        let follows_builder = follows_extensions
-            .get_mut::<otel::SpanBuilder>()
-            .expect("Missing SpanBuilder span extensions");
+        let mut follows_data = follows_extensions
+            .get_mut::<OtelData>()
+            .expect("Missing otel data span extensions");
 
         let follows_context = self
             .tracer
-            .sampled_context(follows_builder)
+            .sampled_context(&mut follows_data)
```

Comment:
nit: clippy points out that the borrow isn't necessary here:
```suggestion
            .sampled_context(follows_data)
```